<launch>
   <group ns="$(env HOSTNAME)">

   <!-- Initialize the Onboard Setpoint Publisher -->
	<node pkg="bridge_px4" type="setpoint_publisher" name ="setpoint_publisher" output="screen">
	   <param name="spo_hz" value="200.0"/>
      <param name="cup_hz" value="200.0"/>
      <param name="spi_hz_min" value="10.0"/>
      <param name="mci_hz_min" value="5.0"/>
      <param name="sa_z" value="1.0"/>
      <param name="x_min" value="-7.8"/>
      <param name="x_max" value="7.8"/>
      <param name="y_min" value="-2.9"/>
      <param name="y_max" value="2.9"/>
      <param name="z_min" value="0.3"/>
      <param name="z_max" value="2.7"/>
	</node>
   
   <!-- Initialize the High Rate Controller -->
   <!-- <node pkg="bridge_px4" type="hr_control" name ="hr_control" output="screen">
      <param name="pxy_slim" value="1.0"/>
      <param name="pz_slim" value="0.5"/>
      <param name="v_slim" value="3.0"/>
      <param name="q_slim" value="1.0"/>
      <param name="ep_lim" value="1.0"/>
      <param name="eq_lim" value="0.5"/>
   </node> -->

   <!-- Initialize the mavros bridge over serial -->
	<include file="$(find mavros)/launch/node.launch">
	   <arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
      <arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

	   <arg name="fcu_url" value="/dev/ttyS4:921600" />
      <arg name="gcs_url" value="" />
      <arg name="tgt_system" value="1" />
      <arg name="tgt_component" value="1" />
      <arg name="log_output" value="screen" />
      <arg name="fcu_protocol" value="v2.0" />
      <arg name="respawn_mavros" default="false" />
	</include>

   </group>

</launch>
